internal/setup/setup.go

```diff
@@ -189,6 +190,30 @@ func (r *Release) validate() error {
 		paths[newPath] = new
 	}
 
+	// Check for archive priority conflicts.
+	priorities := make(map[int]*Archive)
+	for _, archive := range r.Archives {
+		old, ok := priorities[archive.Priority]
+		if !ok {
+			priorities[archive.Priority] = archive
+			continue
+		}
+		if old.Name > archive.Name {
+			archive, old = old, archive
+		}
+		return fmt.Errorf("archives %q and %q have the same priority value of %v", old.Name, archive.Name, archive.Priority)
+	}
+
+	// Check package pinned archive validity.
+	for _, pkg := range r.Packages {
+		if pkg.Archive == "" {
+			continue
+		}
+		if _, ok := r.Archives[pkg.Archive]; !ok {
+			return fmt.Errorf("%s: archive %q not defined", pkg.Path, pkg.Archive)
+		}
+	}
+
```

Comment:
Question: All the archives in `r.Archives` are valid ones, right? Those are only created when reading from yaml and then opening them AFAIK.

Reply:
So, yeah, they should be valid ones. They are parsed from `chisel.yaml` and if any archive is syntactically invalid, it's reported right then. `r.Archives` do not contain opened archives though. Archives are opened once the releases are validated, slices are selected etc.. In that case too, if an archive cannot be opened, an error is returned.

Reply:
Right, so we are checking that the archive is "defined" and then we are going to open all of the archives from chisel.yaml, so that means that all the archives are now going to be valid. Thanks!